py/torch_tensorrt/dynamo/lowering/_remove_sym_nodes.py

```diff
@@ -0,0 +1,30 @@
+import logging
+
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def remove_sym_nodes(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    """Remove sym_int placeholders which get inserted due to torch.compile's
+    dynamic=True behavior
+    """
+    # Extract SymInt placeholder Tensors
+    placeholders = [
+        node
+        for node in gm.graph.nodes
+        if (
+            node.op == "placeholder"
+            and isinstance(node.type, type)
+            and issubclass(node.type, torch.SymInt)
+        )
+    ]
+
+    for node in placeholders:
+        gm.graph.erase_node(node)
```

Comment:
How does this affect downstream consumers of the `SymInt` node, such as tensors having that node as a shape dimension?

Reply:
From the cases that I validated, there seems to be no users of these `SymInt` nodes. If we encounter such case, we need to refactor this pass accordingly.